cluster-agent/utils/argocd_create.go

```diff
@@ -366,20 +383,27 @@ func SetupArgoCD(ctx context.Context, apiHost string, argoCDNamespace string, k8
 			},
 		},
 	}
-
-	if err := k8sClient.Create(ctx, &clusterRole); err != nil {
-		if apierr.IsAlreadyExists(err) {
-			if err := k8sClient.Update(ctx, &clusterRole); err != nil {
-				return fmt.Errorf("error on Update %v", err)
-			}
-			logutil.LogAPIResourceChangeEvent(clusterRole.Namespace, clusterRole.Name, clusterRole, logutil.ResourceModified, log)
-
-		} else {
-			return fmt.Errorf("error on Create %v", err)
+	actualClusterRole := rbac.ClusterRole{}
+
+	err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&clusterRole), &actualClusterRole)
+	if err == nil {
+		// It already exists, so we update it.
```

Comment:
Were you intending to add a unit test in argocd_create_test.go to test this err == nil if condition?  I only see the clusterRoler being created.

Reply:
Good catch, I've updated the unit test.